app/Main.hs

```diff
@@ -244,52 +246,127 @@ findDelimiter (s:ss) =
   then s
   else findDelimiter ss
 
+data Command
+  = CmdParse
+  | CmdFeedback Bool
+  | CmdCheck
+  | CmdValid
+  | CmdShow
+  | CmdPending (Maybe Int)
+  | CmdStatus (Maybe Int)
+  | CmdSummary (Maybe Word)
+  | CmdExport (Maybe String)
+  | CmdExportResults
+  | CmdExportTable
+  | CmdExportHTML
+  | CmdExportMD
+  | CmdExportPdfMark
+  deriving (Show)
+
+data Args = Args
+  { argsCommand :: Command
+  , argsPaths :: [String]
+  }
+  deriving (Show)
+
+cmdFeedbackParser :: App.Parser Command
+cmdFeedbackParser =
+  CmdFeedback
+    <$> switch
+      ( long "with-points" )
+  <**> helper
+
+cmdWithDepthParser :: Read a => (Maybe a -> Command) ->  App.Parser Command
+cmdWithDepthParser cmd =
+  cmd
+    <$> (optional $ option auto
+      ( long "depth" ))
+  <**> helper
+
+cmdExportParser :: App.Parser Command
+cmdExportParser =
+  CmdExport
+    <$> (optional $ strOption
+      ( long "format" ))
+  <**> helper
+
+argsParser :: App.Parser Args
+argsParser = Args
+  <$> subparser
+    ( command "parse" (info (pure CmdParse) (progDesc
+        "parse the given PATHs and show their ASTs"))
+   <> command "feedback" (info cmdFeedbackParser (progDesc
+        "accumulate feedback for PATHs"))
+   <> command "valid" (info (pure CmdValid) (progDesc
+        "validate structure of files at PATHs"))
+   <> command "check" (info (pure CmdCheck) (progDesc
+        "validate and check that points have been given at PATHs"))
+   <> command "show" (info (pure CmdShow) (progDesc
+        "validate, check, and pretty-print"))
+   <> command "pending" (info (cmdWithDepthParser CmdPending) (progDesc
+        "show judgements not yet marked"))
+   <> command "status" (info (cmdWithDepthParser CmdStatus) (progDesc
+        "show status(?)"))
```

Comment:
There are too many conflating terms here. "valid", "check", "status". I think just one should really suffice, or some clear explanation as to the difference is in order. Again, this might be subject matter for another PR.

Reply:
Maybe the solution is to group the commands into those you use frequently while working (`status`) and those you only use when you are setting up a marking structure initially (`valid`).